cmd/abad/cli/cli.go

```diff
@@ -0,0 +1,62 @@
+package cli
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+
+	"github.com/NeowayLabs/abad"
+)
+
+type (
+	Cli struct {
+		in  io.Reader
+		out io.Writer
+
+		js *abad.Abad
+	}
+)
+
+func NewCli(fname string, in io.Reader, out io.Writer) *Cli {
+	return NewWithJS(abad.NewAbad(fname), in, out)
+}
+
+func NewWithJS(js *abad.Abad, in io.Reader, out io.Writer) *Cli {
+	return &Cli{
+		in:  in,
+		out: out,
+		js:  js,
+	}
+}
+
+func (c *Cli) ReadEval() {
+	fmt.Fprintf(c.out, "> ")
+	bio := bufio.NewReader(c.in)
+	line, err := bio.ReadString('\n')
+	if err != nil {
+		c.errorf(err)
+		return
+	}
+
+	if len(line) > 0 {
```

Comment:
What you are doing here would be a strings.TrimSuffix(line, "\n") ?

Reply:
yes, but does it makes sense to use it to just drop the last char? `TrimSuffix` involves calling `HasSuffix` but we know the suffix is there. I can make a function called `trimnl` that just drops the new line at end.

Also, the `bio.ReadString('\n')`  always returns at least the delimiter, then the `if check` above could be dropped also.